### ISSUE TITLE
Hitboxes cause unexpected collisions

### ISSUE THREAD

**Issue by RahlForge** [Raptor007/X-Wing-Revival#4]
The hitboxes on the starfighters and asteroids are spherical, which often causes unexpected collisions when one might expect to be "skimming the surface". This is especially prevalent in the Battle of Yavin, where you can accidentally clip your fighter on trench projections you should have missed (underside/topside for horizontally-oriented craft like X-Wings and Y-Wings, and starboard/port sides for more vertically-oriented craft like B-Wings and TIE Fighters), and battles with Many+ asteroids.

**Comment by Raptor007** [comment]
Finally getting somewhere with this!  I started by making [simple collision models for fighters](https://i.imgur.com/PFiOIfN.png) but that tanked server performance and didn't even work correctly -- they'd just pass right through capital ships.  So I went digging into the collision detection code to fix and optimize it.  Now I finally understand some of the odd glitches I've seen, such as capital ships occasionally passing through each other and other times colliding before actually touching.

Brief overview of collision detection between models:
1. Filter down to pairs of game objects that are allowed to collide with each other and near enough to do it this frame.
2. For each pair, divide space into blocks and mark each block with triangles from either model that pass through it.
3. For each block marked by both models, see if any triangles intersect (or will as they move).

This little piece turned out to be a huge part of the problem:
https://github.com/Raptor007/RaptorEngine/blob/c4e029a2ac84457db2e0bd0529301748dffe79c4/Graphics/Model.cpp#L886-L887

In addition to not considering the motion of either ship, that second line should have been:
`model2->MarkBlockMap( &blockmap2, &arrays2, pos2, object_names2, exploded2, explosion_seed2, block_size );`

So the blockmap step was using the same ship model twice instead of using each ship's model for its own position!  Whoops.

While trying to chase down this bug I did a lot of rewriting throughout the collision detection code, so by the time I'd managed to actually figure out what was wrong, I'd already refactored most of the sections that were hampering performance.  (In complete reverse from common programming wisdom, I made it fast *then* made it work, haha!)
